catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/FeedbackApplication.java

```diff
@@ -0,0 +1,264 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query;
+
+import static spark.Spark.exception;
+import static spark.Spark.post;
+import static spark.route.RouteOverview.enableRouteOverview;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.shiro.SecurityUtils;
+import org.boon.core.value.ValueList;
+import org.boon.json.JsonFactory;
+import org.boon.json.ObjectMapper;
+import org.codice.ddf.catalog.ui.config.ConfigurationApplication;
+import org.codice.ddf.catalog.ui.query.feedback.FeedbackRequest;
+import org.codice.ddf.platform.email.SmtpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+
+import ddf.security.Subject;
+import ddf.security.SubjectUtils;
+
+import spark.servlet.SparkApplication;
+
+public class FeedbackApplication implements SparkApplication {
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FeedbackApplication.class);
+
+    private static final ObjectMapper MAPPER = JsonFactory.create();
+
+    private ConfigurationApplication configurationApplication = null;
+
+    private Template subjectTemplate = null;
+
+    private Template bodyTemplate = null;
+
+    private SmtpClient smtpClient = null;
+
+    private String emailDestination = null;
+
+    @Override
+    public void init() {
+        post("/feedback", APPLICATION_JSON, (req, res) -> {
+            try {
+                FeedbackRequest feedback = parseFeedbackRequest(req.body());
+                feedback.setAuthUsername(getCurrentUser());
+
+                String emailSubject = getEmailSubject(feedback);
+                String emailBody = getEmailBody(feedback);
+                emailBody = emailBody.replaceAll("\\\\n", "\n");
+
+                Session emailSession = smtpClient.createSession();
+                MimeMessage message = new MimeMessage(emailSession);
+                message.addRecipient(Message.RecipientType.TO, new InternetAddress(emailDestination));
+                message.setSubject(emailSubject);
```

Comment:
Per http://docs.oracle.com/javaee/6/api/javax/mail/internet/MimeMessage.html#setSubject(java.lang.String) I think null would be ok, as it would just not set the subject.